### ISSUE TITLE
Twitch.tv Integration

### ISSUE THREAD

**Issue by Tecm0** [SoundControl/Issues#139]
Twitch.tv Integration

**Comment by SoundControl** [comment]
Yes. Twitch.tv looks tricky, but doable.

**Comment by wahayoyo** [comment]
Also interested in twitch integration, mainly just pause/mute when another music source is active.

**Comment by Poprox198** [comment]
Any update on this integration?

**Comment by SndControl** [comment]
Poprox198,

We have not begun the Twitch.tv Integration. We are receiving many new
integration requests because of the new services coming online like
TidalHiFi and Deezer coming to US, and of course Twitch.tv.

We are currently working on developing new features first and then adding
support for new services, since our first focus was on integration. We can
use your help in arguing for Twitch.tv to be the first new integration to
the development team and getting some new features now that will be good
for Twitch.tv when we switch to an integration cycle.

So, two quick questions:
1. How (or for what) and when do you use Twitch.tv in the Chrome browser?
2. And, if any, what new features could we add that would help the
integration of Twitch.tv be seamless?

If you have time, we would love to hear about your opinion on these two
matters because we will have to relay this message to management (and find
we management listens better to support opinions if we support them with
actual user feedback, rather than the Support Team's interpretation of user
feedback).

Thanks!

Clay
SoundControl Support Team
support@sndcontrol.com - email
Follow us @sndcontrol (DM or #SndHelps for inquiries) - Twitter

On Thu, Jan 29, 2015 at 1:26 PM, Poprox198 notifications@github.com wrote:

> Any update on this integration?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/SoundControl/Issues/issues/139#issuecomment-72087508.

## 

Clayton Chamberlain | SndControl
(918) 644-0639 - Mobile
clayton@sndcontrol.com - Email
http://sndcontrol.com - Visit

**Comment by Poprox198** [comment]
Clay,
1. I use twitch TV to watch professional gaming events for 20+ hours a
   week.
2. Features that I am looking for are: The ability to mute streams and move
   between tabs that are running twitch.

Sometimes I will be working and listening to music on my google music
account and watching twitch. Your application allows me to control google
music just fine, hotkeys for controlling instance volume would be nice, but
thats a separate issue... The point is that if I want to listen to music
and watch twitch muted I have to manually move between pages, rather than a
quick hotkey to "move the focus" so to speak.

There are other extensions to mute pages, but I like the idea of having a
central one to bind to my standard music hotkeys.

In short, moving between services and selectively controlling volume in
each tab.

I also have done a little bit of introductory work in google script so if
you guys have any open development available on GitHub I wouldn't mind
contributing if this needs help.

Thanks,

Josh

**Comment by SndControl** [comment]
Josh,

Great. I am meeting with our head developer Scott this weekend and I will
run this by him. What is your email so we can email you directly after we
talk and we can also add you to our Google Group testing when we start
releasing some new features for you to test out with a critical eye. Thanks
Josh.

Clay
support@sndcontrol.com (support team)
clayton@sndcontrol.com (work)

**Comment by Poprox198** [comment]
Kaminskyjoshua@Gmail.com

**Comment by phit** [comment]
would love to see this aswell!

> So, two quick questions:
> 1. How (or for what) and when do you use Twitch.tv in the Chrome browser?
> 2. And, if any, what new features could we add that would help the
> integration of Twitch.tv be seamless?
1. watch random streams all day mainly in the background
2. an option to either pause the stream or just mute it when another media source is active

thanks for all the good work so far

**Comment by SndControl** [comment]
No problem, we love what we do and we love more working with our users to
get fun new things created (as our user base is really creative and tech
savvy).

Thanks for letting us know phit! We are building our website now to open up
the ability to add new features without overloading the extension because
we want to stay as simple and lean as possible in the extension for the
sake of our users and our goal of providing more listening power with fewer
distractions. We are a small team and want to get it all done as fast as we
can, but we are looking to grow this year as well and hopefully get more
help to respond to your requests quicker. For transparency sake and to let
you know we haven't forgot this, it looks like the end of March will be
when we will be doing a new integration cycle.

Do you have any expertise in chrome extensions or web apps? Like with Josh
(who pointed out this issue), if you have any other good ideas on this
integration, you can email me at clayton@sndcontrol.com.

Clay | SndControl
SoundControl Support Team
clayton@sndcontrol.com (personal)
support@sndcontrol.com (work)
@sndcontrol (twitter & github)

**Comment by phit** [comment]
Hey Clay,

I can understand that, a modular system like e.g. [proxmate](http://proxmate.me/) uses would be pretty cool. You can look at their implementation for some inspiration, it's open-sourced and on [github](http://proxmate.me/about/).
Anyway it sounds great and I wish you guys all the best and can't wait for those updates!

I do have some experience with chromium extensions, but currently not enough free time to help out besides testing. Generally for integration the proxmate implementation mentioned above should give you guys a good starting point, at least from a concept standpoint.

phit@phit.link
